File: code/gms-backend/src/main/java/io/github/gms/secure/service/SystemPropertyService.java
package io.github.gms.secure.service;

import io.github.gms.common.enums.SystemProperty;
import io.github.gms.secure.dto.PagingDto;
import io.github.gms.secure.dto.SystemPropertyDto;
import io.github.gms.secure.dto.SystemPropertyListDto;

/**
 * @author Peter Szrnka
 * @since 1.0
 */
public interface SystemPropertyService {
	
	void save(SystemPropertyDto dto);
	
	void delete(String key);
	
	SystemPropertyListDto list(PagingDto dto);

	String get(SystemProperty key);
	
	Long getLong(SystemProperty key);

	boolean getBoolean(SystemProperty key);

	Integer getInteger(SystemProperty key);
}

File: code/gms-frontend/src/app/components/system_property/service/system-property.service.ts
import { HttpClient } from "@angular/common/http";
import { Injectable } from "@angular/core";
import { environment } from "../../../../environments/environment";
import { Observable } from "rxjs";
import { SystemProperty } from "../model/system-property.model";
import { getHeaders } from "../../../common/utils/header-utils";
import { Paging } from "../../../common/model/paging.model";
import { SystemPropertyList } from "../model/system-property-list.model";

/**
 * @author Peter Szrnka
 */
@Injectable()
export class SystemPropertyService {

    constructor(protected http : HttpClient) {}

    save(item : SystemProperty) : Observable<string> {
        return this.http.post<string>(environment.baseUrl + 'secure/system_property', item, { withCredentials: true, headers : getHeaders() });
    }

    delete(key: string): Observable<string> {
        return this.http.delete<string>(environment.baseUrl + 'secure/system_property/' + key, { withCredentials: true, headers : getHeaders() });
    }

    list(paging: Paging): Observable<SystemPropertyList> {
        return this.http.get<SystemPropertyList>(environment.baseUrl + 'secure/system_property/list' + 
            `?direction=${paging.direction}&property=${paging.property}&page=${paging.page}&size=${paging.size}`, 
            { withCredentials: true, headers : getHeaders() });
    }
}